pkg/loopback/ioctl.go

```diff
@@ -9,11 +9,15 @@ import (
 )
 
 func ioctlLoopCtlGetFree(fd uintptr) (int, error) {
-	index, err := unix.IoctlGetInt(int(fd), LoopCtlGetFree)
-	if err != nil {
+	// The ioctl interface for /dev/loop-control (since Linux 3.1) is a bit
+	// off compared to what you'd expect: instead of writing an integer to a
+	// parameter pointer like unix.IoctlGetInt() expects, it returns the first
+	// available loop device index directly.
```

Comment:
Should / could `golang.org/x/sys` be changed to take that into account? Or not generic enough to do so? 

/cc @tklauser

Reply:
@thaJeztah @kolyshkin if this is not just a special case for `LOOP_CTL_GET_FREE` and there are other possible cases, I think it's worthwhile to add something like `IoctlRetInt` to `x/sys/unix`. Feel free to send a CL/PR and add me as a reviewer.

Reply:
@tklauser filed https://github.com/golang/go/issues/33966; working on implementation

Reply:
https://go-review.googlesource.com/c/sys/+/192780